### ISSUE TITLE
grad_norm is nan

### ISSUE THREAD

**Issue by FilippoBotti** [huangzizheng01/ShuffleMamba#2]
I've tried to run your script without any change but the grad_norm is always nan and the model does not train. Any help?

`
Namespace(ThreeAugment=False, aa='rand-m9-mstd0.5-inc1', accum_iter=1, amp='bfloat16', batch_size=128, bce_loss=False, clip_grad=3.0, color_jitter=0.3, cooldown_epochs=10, cutmix=1.0, cutmix_minmax=None, data_path='ImageNet100', data_set='IMNET', decay_epochs=30, decay_rate=0.1, device='cuda', dist_backend='nccl', dist_eval=True, dist_url='env://', distributed=True, drop=0.0, drop_path=0.5, epochs=300, eval=False, eval_crop_ratio=0.875, finetune='', gpu=0, inat_category='name', input_size=224, local_rank=0, lr=0.0005, lr_noise=None, lr_noise_pct=0.67, lr_noise_std=1.0, min_lr=5e-06, mixup=0.8, mixup_mode='batch', mixup_prob=1.0, mixup_switch_prob=0.5, model='mambar_base_patch16_224', model_ema=True, model_ema_decay=0.9999, model_ema_force_cpu=False, momentum=0.9, num_workers=4, opt='adamw', opt_betas=[0.9, 0.95], opt_eps=1e-08, output_dir='results/basic', patience_epochs=10, pin_mem=True, rank=0, recount=1, remode='pixel', repeated_aug=True, reprob=0.25, resplit=False, resume='', save_freq=50, sched='cosine', seed=0, smoothing=0.1, src=False, ssr=0.5, start_epoch=0, train_interpolation='bicubic', train_mode=True, unscale_lr=False, warmup_epochs=30, warmup_lr=5e-07, weight_decay=0.1, world_size=1)
Creating model: mambar_base_patch16_224
number of params: 97598440
Start training for 300 epochs
Epoch: [0]  [  0/390]  eta: 1:13:58  lr: 0.000000  loss: 7.0813 (7.0813)  grad_norm: nan (nan)  time: 11.3805  data: 3.2103  max mem: 19859
`

**Comment by huangzizheng01** [comment]
Hi, thanks for your interest in our repo and sorry for replying late. I just tried your settings on ImageNet1K, i.e., use 1 gpu and the batchsize of 128, but did not reproduce your issue. Here is the printed lines:

Start training for 300 epochs
Epoch: [0]  [    0/10008]  eta: 1 day, 12:48:58  lr: 0.000000  loss: 6.9789 (6.9789)  grad_norm: 21.5764 (21.5764)  time: 13.2432  data: 2.3202  max mem: 19536
Epoch: [0]  [   10/10008]  eta: 5:57:38  lr: 0.000000  loss: 7.0298 (7.0492)  grad_norm: 21.0658 (20.9522)  time: 2.1463  data: 0.2111  max mem: 20694
Epoch: [0]  [   20/10008]  eta: 4:27:28  lr: 0.000000  loss: 7.0330 (7.0431)  grad_norm: 20.5437 (20.8067)  time: 1.0250  data: 0.0002  max mem: 20694

As the [0/390] progress bar provided by your information, I guess you are experimenting with a much smaller dataset.

**Comment by FilippoBotti** [comment]
Thanks for your answer, I forgot to mention that I am using ImageNet100 (only 100 classes of imagenet) in order to study the code and run it on my 4090. Do you have any suggest on which hyperparameters I have to change in order to run on a dataset like this?

**Comment by huangzizheng01** [comment]
My validation is primarily based on IN-1K, so I’m afraid I don’t have much direct insight into your experiments—sorry about that! That said, you could try tweaking some key hyperparameters, such as the learning rate, ssr, and drop_path, to see if that improves the results. Thanks!

**Comment by FilippoBotti** [comment]
Thanks for you help :)
